ceremonies/src/lib.rs

```diff
@@ -359,148 +414,326 @@ decl_error! {
+        let assignment_params_endorsees = Self::generate_assignment_function_params(num_assigned_endorsees, num_meetups);
+        let assignment_params_newbies = Self::generate_assignment_function_params(num_assigned_newbies, num_meetups);
+
+        AssignmentParamsBootstrappersReputables::insert(community_ceremony, assignment_params_bootstrappers_reputables);
+        AssignmentParamsEndorsees::insert(community_ceremony, assignment_params_endorsees);
+        AssignmentParamsNewbies::insert(community_ceremony, assignment_params_newbies);
+
+        <AssignedBootstrapperCount>::insert(community_ceremony, num_assigned_bootstrappers);
+        <AssignedReputableCount>::insert(community_ceremony, num_assigned_reputables);
+        <AssignedEndorseeCount>::insert(community_ceremony, num_assigned_endorsees);
+        <AssignedNewbieCount>::insert(community_ceremony, num_assigned_newbies);
+        <MeetupCount>::insert(community_ceremony, num_meetups);
+        Ok(())
+    }
+
+    fn generate_all_meetup_assignment_params() -> DispatchResult{
         let cids = <encointer_communities::Module<T>>::community_identifiers();
         let cindex = <encointer_scheduler::Module<T>>::current_ceremony_index();
+        for cid in cids.iter() {
+            Self::generate_meetup_assignment_params((*cid, cindex)).ok();
+        }
+        Ok(())
+    }
+
+    fn is_prime(n: u64) -> bool {
+        if n <= 3 {
+            return n > 1;
+        }
+        if n % 2 == 0 || n % 3 == 0 {
+            return false;
+        }
+        if n < 25 {
+            return true;
+        }
+        let mut i: u64 = 5;
+        while i.pow(2) <= n {
+            if n % i == 0u64 || n % (i + 2u64) == 0u64 {
+                return false;
+            }
+            i += 6u64;
+        }
+        return true;
+    }
+
+
+    fn find_prime_below(mut n: u64) -> u64 {
+        if n <= 2 {
+            return 2u64;
+        }
+        if n % 2 == 0 {
+            n -= 1;
+        }
+        while n > 0 {
+            if Self::is_prime(n) {
+                return n;
+            }
+            n -= 2;
+        }
+        2u64
+    }
 
+
+    fn mod_inv(a: i64, module: i64) -> i64 {
+        let mut mn = (module, a);
+        let mut xy = (0, 1);
+
+        while mn.1 != 0 {
+            xy = (xy.1, xy.0 - (mn.0 / mn.1) * xy.1);
+            mn = (mn.1, mn.0 % mn.1);
+        }
+
+        while xy.0 < 0 {
+            xy.0 += module;
+        }
+        xy.0
+    }
+
+    fn validate_equal_mapping(num_participants: u64, m: u64, n: u64, s1: u64, s2:u64) -> bool{
+        if num_participants < 2 {
+            return true;
+        }
+
+        let mut meetup_index_count: Vec<u64> = vec![0; n as usize];
+        let mut meetup_index_count_max = (num_participants - m) / n;
+        if (num_participants - m) % n != 0 {meetup_index_count_max += 1;}
```

Comment:
same here: using a ceil function gives you immutability of a variable.. this is a very good trade.. ;)

Reply:
noob question: why is it desirable for a variable to be immutable?

Reply:
philosophical approach: changing the value of a variable is achieved by either doing some mathematical operation on it or by plainly overwriting it.. in both cases the meaning of the value changes. take the physical world as an example.. let's say we talk about my weight.. 
1. you put me on a scale and read the displayed mass in kg. if you substract the weight of my clothes and store into the same variable, you are inconsistent with what you describe as my weight. is it naked weight or clothed weight. any programmer debugging any function part of code that treats "weight" will have to figure out, which type of weight you meant.. 
2. you put yourself on the same scale and overwrite my weight.. same argument as above.. I will always have to figure out which weight I am working at

I'm sure there are better examples and explanations. also you will find much better literature on it by simply searching for immutability of objets/variables etc online. for example: if you only work with immutable variables, you will be able to parallelize any loop without side-effect (and if side-effects exist, apply this rule recursively, except execution order matters)